src/Microsoft.DotNet.Darc/DarcLib/Models/VirtualMonoRepo/ManifestRecord.cs

```diff
@@ -27,19 +28,27 @@ public string GetPublicUrl()
     {
         var url = RemoteUri;
 
-        if (url.EndsWith(".git"))
+        switch (GitRepoUrlParser.ParseTypeFromUri(url))
```

Comment:
This logic should really be unit tested, especially when there are string manipulations.

Reply:
@tkapin added tests